### ISSUE TITLE
Remove unnecessary null pointer checks

### ISSUE THREAD

**Issue by elfring** [libimobiledevice/libirecovery#7]
[An extra null pointer check is not needed in functions](http://pubs.opengroup.org/onlinepubs/9699919799/functions/free.html) like the following.
- [irecv_close](https://github.com/libimobiledevice/libirecovery/blob/3687ea11271b79fb2643c8f041490544c17c5e58/src/libirecovery.c#L993)
- [irecv_execute_script](https://github.com/libimobiledevice/libirecovery/blob/3687ea11271b79fb2643c8f041490544c17c5e58/src/libirecovery.c#L1397)

Would you like to apply a [semantic patch](http://coccinelle.lip6.fr/) like the following to find more update candidates?

```
@Remove_unnecessary_pointer_checks@
expression x;
@@
-if (x)
    free(x);
```

**Comment by nikias** [comment]
Fixed with commit f78fc4a11b81bcb2aacf93a2060f15fa2f99b66a.

**Comment by elfring** [comment]
Thanks for [your small source code improvement](https://github.com/libimobiledevice/libirecovery/commit/f78fc4a11b81bcb2aacf93a2060f15fa2f99b66a "Remove some unnecessary null pointer checks").
